drivers/storage/portworx/components_test.go

```diff
@@ -3128,6 +3120,42 @@ func TestSecurityInstall(t *testing.T) {
 			},
 		},
 	}
+	validateAuthSecurityInstall(t, cluster)
+
+	// validate with security disabled, but auth enabled
+	cluster = &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1.StorageClusterSpec{
+			Security: &corev1.SecuritySpec{
+				Enabled: false,
```

Comment:
I think in this case, security is disabled, and therefore TLS should also be disabled.

Parent enabled should take precedence over child enabled.